clang/test/CodeGenSYCL/sycl-add-opt-level-attrib.cpp

```diff
@@ -0,0 +1,18 @@
+// RUN: %clangxx %s -O0 -S -o %t.ll -fsycl-device-only
```

Comment:
Do we need an additional test with explicit `__attribute__((optnone))` on the kernel/device function?

Reply:
My changes do not interact with this attribute. The backend should be responsible to decide whether the attribute or the option takes precedence.

Thanks